app/epics/experimentEpics.js

```diff
@@ -149,7 +149,7 @@ const experimentStopEpic = (action$, state$) =>
         state$.value.experiment.group,
         state$.value.experiment.session
       );
-      if (state$.device.deviceType === DEVICES.EMOTIV) {
+      if (state$.value && state$.value.device && state$.value.device.deviceType && state$.value.device.deviceType === DEVICES.EMOTIV) {
```

Comment:
out of curiosity, what led to all three being needed vs. one?

Reply:
It's just to check whether each of these objects exist before checking the device type. As far as I know, it would produce an error if something is missing, like for example, there is no "device" in value. 
But I guess this should be already checked elsewhere, so we can just use _state$.value.device.deviceType === DEVICES.EMOTIV_ 
In a new update, I also added a check whether EEG enabled so it does not try to stop recording in behavioral experiments.